From f82750e615166b05d479e02de170f375f64af321 Mon Sep 17 00:00:00 2001
From: David Wu <david.wu@rock-chips.com>
Date: Mon, 29 Oct 2018 10:55:46 +0800
Subject: [PATCH] pinctrl: rockchip: Call rockchip_pinctrl_get_soc_data() after
 regmap Initialization

In soc_data_init(), need to write GRF registers, so change
the order for rockchip_pinctrl_get_soc_data(), call it after
regmap Initialization.

Change-Id: Ia3a6a821861d853c85870c2be43f33c6ad76c0e9
Signed-off-by: David Wu <david.wu@rock-chips.com>
---
 drivers/pinctrl/pinctrl-rockchip.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/pinctrl/pinctrl-rockchip.c b/drivers/pinctrl/pinctrl-rockchip.c
index ec5fe39de6ea..f6a340e05a60 100644
--- a/drivers/pinctrl/pinctrl-rockchip.c
+++ b/drivers/pinctrl/pinctrl-rockchip.c
@@ -4413,13 +4413,6 @@ static int rockchip_pinctrl_probe(struct platform_device *pdev)
 
 	info->dev = dev;
 
-	ctrl = rockchip_pinctrl_get_soc_data(info, pdev);
-	if (!ctrl) {
-		dev_err(dev, "driver data not available\n");
-		return -EINVAL;
-	}
-	info->ctrl = ctrl;
-
 	node = of_parse_phandle(np, "rockchip,grf", 0);
 	if (node) {
 		info->regmap_base = syscon_node_to_regmap(node);
@@ -4463,6 +4456,13 @@ static int rockchip_pinctrl_probe(struct platform_device *pdev)
 			return PTR_ERR(info->regmap_pmu);
 	}
 
+	ctrl = rockchip_pinctrl_get_soc_data(info, pdev);
+	if (!ctrl) {
+		dev_err(dev, "driver data not available\n");
+		return -EINVAL;
+	}
+	info->ctrl = ctrl;
+
 	ret = rockchip_gpiolib_register(pdev, info);
 	if (ret)
 		return ret;
-- 
2.35.3

